src/core/utils/geometryutils.cpp

```diff
@@ -133,3 +134,20 @@ double GeometryUtils::distanceBetweenPoints( const QgsPoint &start, const QgsPoi
 {
   return start.distance( end );
 }
+
+QgsPoint GeometryUtils::reprojectPointToWgs84( const QgsPoint &point, const QgsCoordinateReferenceSystem &crs )
+{
+  QgsCoordinateReferenceSystem wgs84Crs = QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4326" ) );
+  const QgsCoordinateTransform ct( crs, wgs84Crs, QgsProject::instance() );
+  QgsPointXY reprojectedPoint;
+  try
+  {
+    ct.transform( point.x(), point.y() );
```

Comment:
Height is lost here. Ok? Needs docs?

Reply:
@m-kuhn , na you're right we should insure Z and M values stick if present. Good catch.